### ISSUE TITLE
test for relative_eff with multiple cores fails on windows 

### ISSUE THREAD

**Issue by jgabry** [stan-dev/loo#129]
There's one test that's failing only on windows (appveyor) and not on linux (travis) or my mac.
@paul-buerkner Whenever you have a chance would you be able to check if this this test also fails for you on Windows? (I assume you're still on Windows) 

Here's the info from appveyor (https://ci.appveyor.com/project/jgabry/loo/builds/28868551#L1231): 

```
Last 13 lines of output:
  2 nodes produced errors; first error: object 'draws' not found
  Backtrace:
   1. loo::relative_eff(...)
   2. loo:::relative_eff.function(...)
   3. parallel::parLapply(...)
   5. parallel::clusterApply(...)
   6. parallel:::staticClusterApply(cl, fun, length(x), argfun)
   7. parallel:::checkForRemoteErrors(val)
  
  == testthat results  ===========================================================
  [ OK: 990 | SKIPPED: 0 | WARNINGS: 0 | FAILED: 1 ]
  1. Error: relative_eff with multiple cores runs (@test_relative_eff.R#44) 
  
  Error: testthat unit tests failed
  Execution halted
```

**Comment by paul-buerkner** [comment]
Yes, it still fails for me.

**Comment by jgabry** [comment]
Thanks. Any idea what’s going wrong on Windows? Multiple cores seems to work on Windows for everything but relative_eff seemingly

**Comment by paul-buerkner** [comment]
Windows seems to be the only OS where `parallel::parLapply` is used. Does the same error occur when this function is used on Mac?

**Comment by jgabry** [comment]
Good point! It does fail for me too if I change it to use parLapply on mac too. It seems to fix it if I add

```r
parallel::clusterExport(cl, "draws")
```

but it's weird that this didn't seem to be necessary previously.

**Comment by bachlaw** [comment]
I assume that this relates to the fact that mc.cores is a forking method and that Windows therefore cannot use it.  When I try to run `loo` in windows with a `cores` argument, I get the following error message: `Error in get(name, envir = envir) : object 'draws' not found`, which sounds quite similar to what the fix @jgabry highlights would address.  This is on version 2.2 for Windows; if there is a different version worth trying, happy to do so.

**Comment by bgoodri** [comment]
This is still messed up and is causing a rstanarm example to fail. That line should be
```
parallel::clusterExport(cl, "draws", envir = environment())
```
or else it does not find `draws` in the default `.GlobalEnv`.

**Comment by rok-cesnovar** [comment]
This was fixed 2 days ago: https://github.com/stan-dev/loo/pull/152 
Or is it an issue even with that patch.

**Comment by jgabry** [comment]
Yeah @bgoodri should be fixed by #152, but it's not on CRAN yet. Do you need it on CRAN before submitting rstanarm? We could also use relative_eff matrix method instead of function method (at least temporarily) in rstanarm.

**Comment by bgoodri** [comment]
It is better to change the rstanarm example to pass cores = 1 if we have to
upload a rstanarm before loo. But is loo waiting on anything specific?

**Comment by jgabry** [comment]
loo isn't waiting on anything specific, I've just already done two CRAN releases of loo this month (because of the bug in `loo_moment_match()`) and this is the only change since then. So I was waiting a little while to do another CRAN submission. Do you think I should just go ahead and submit again? Either way, I don't think we should wait on loo to submit rstanarm (we need get rstanarm out right away given the particular issues the current release has). So even if I submit loo I think you should just change the rstanarm example to 1 core and go ahead and submit so that way we don't have to worry about the order of the submissions.

**Comment by bgoodri** [comment]
OK. I will put cores = 1 in the example for now.
